pkg/log/zap/zap.go

```diff
@@ -207,3 +208,39 @@ func NewRaw(opts ...Opts) *zap.Logger {
 	log = log.WithOptions(o.ZapOpts...)
 	return log
 }
+
+// BindToFlagSet func allows CLI passed flags to be parsed.
+func (o *Options) BindFlags(fs *pflag.FlagSet) {
+	// Set Development mode value
+	fs.BoolVar(&o.Development, "zap-devel", false, "Enable zap development mode (changes defaults to console encoder, debug log level, and disables sampling)")
+
+	// Set Encoder value
+	var encVal encoderFlag
+	encVal.setFunc = func(fromFlag zapcore.Encoder) {
+		o.Encoder = fromFlag
+	}
+	fs.Var(&encVal, "zap-encoder", "Zap log encoding ('json' or 'console')")
+
+	// Set the Log Level
+	var levelVal levelFlag
+	levelVal.setFunc = func(fromFlag zap.AtomicLevel) {
+		o.Level = &fromFlag
+	}
+	fs.Var(&levelVal, "zap-log-level", "Zap log level (one of 'debug', 'info', 'error' or any integer value > 0)")
+
+	// Set the StrackTrace Level
+	var stackVal stackTraceFlag
+	stackVal.setFunc = func(fromFlag zap.AtomicLevel) {
+		o.StacktraceLevel = &fromFlag
+	}
+	fs.Var(&stackVal, "zap-stacktrace-level", "Zap log level (one of 'warn',  'error'")
+
+}
+
+// UseFlagOptions to set logger with CLI passed flags.
+func UseFlagOptions(in *Options) Opts {
+	return func(o *Options) {
+		*o = *in
+		o.addDefaults()
```

Comment:
Any reason to call `o.addDefaults()` here when it will be called in [`NewRaw()`](https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/log/zap/zap.go#L200) as well?

Reply:
For example: from operator-SDK, we plan to call the function UseFlagOptions() directly.
See: https://github.com/learnoperators/memcached-operator/blob/master/cmd/manager/main.go#L66

Reply:
`zap.New()` calls `zap.NewRaw()` which calls `o.addDefaults()` after applying the flag options. 
So it ends up being the same thing.
So not sure why we'd call it here before that.